cli/defenseclaw/inventory/agent_discovery.py

```
@@ -818,53 +186,310 @@ class _AgentSpec(NamedTuple):
+    versions_dir = os.path.join(base, "versions")
+    out: list[tuple[str, str]] = []
+    if os.path.isdir(versions_dir):
+        # No `current` pointer: return one tuple per candidate rather
+        # than pre-picking with a local sort key. The naive digit-prefix
+        # sort here would put "3.0.0-alpha" ahead of the stable "2.1.144"
+        # and starve pick_highest_supported (the authoritative
+        # comparator that understands prerelease tails) of the stable
+        # entry.
+        for name in os.listdir(versions_dir):
+            if name and name[0].isdigit():
+                out.append((f"{versions_dir}/{name}", name))
```

Ruff (0.16.0)

[warning] Use `list.extend` to create a transformed list

(PERF401)